middle_end/flambda2/to_cmm/to_cmm_expr.ml

```diff
@@ -33,14 +33,20 @@ end
 (* Bind a Cmm variable to the result of translating a [Simple] into Cmm. *)
 
 let bind_var_to_simple ~dbg env res v ~num_normal_occurrences_of_bound_vars s =
-  let defining_expr, env, res, effects_and_coeffects_of_defining_expr =
-    C.simple ~dbg env res s
-  in
-  let env, res =
-    Env.bind_variable env res v ~effects_and_coeffects_of_defining_expr
-      ~defining_expr ~num_normal_occurrences_of_bound_vars
-  in
-  env, res
+  match Simple.must_be_var s with
+  | Some (alias_of, _coercion) when Flambda_features.classic_mode () ->
```

Comment:
This should not be restricted to classic mode, it can be useful outside of classic mode.

Reply:
Sure, I just didn't want to break too much at the moment.